src/Pretzel.Tests/Templating/Context/DataTests.cs

```diff
@@ -0,0 +1,240 @@
+
+                var template = Template.Parse(@"{{ data.person.name }}");
+
+                var hash = Hash.FromAnonymousObject(new
+                {
+                    Data = data
+                });
+
+                var result = template.Render(hash);
+
+                Assert.Equal("Eric Mill", result.Trim());
+            }
+
+            [Theory]
+            [InlineData("yml", @"name: Eric Mill
+address:
+  street: Some Street
+  postalcode: 1234")]
+            [InlineData("json", @"{
+  name: 'Eric Mill',
+  address: {
+  street: 'Some Street',
+  postalcode: 1234
+  }
+}")]
+            [InlineData("csv", @"name,address.street,address.postalcode
+""Eric Mill"",""Some Street"",1234")]
+            [InlineData("tsv", @"name	address.street	address.postalcode
+""Eric Mill""	""Some Street""	1234")]
+            public void renders_deep_nested_object(string ext, string fileContent)
+            {
+                fileSystem.AddFile(Path.Combine(dataDirectory, $"person.{ext}"), new MockFileData(fileContent));
+
+                var template = Template.Parse(@"{{ data.person.address.postalcode }}");
+
+                var hash = Hash.FromAnonymousObject(new
+                {
+                    Data = data
+                });
+
+                var result = template.Render(hash);
+
+                Assert.Equal("1234", result.Trim());
+            }
+
+            [Theory]
+            [InlineData("yml", @"- name: Eric Mill
+  github: konklone
+
+- name: Parker Moore
+  github: parkr
+
+- name: Liu Fengyun
+  github: liufengyun")]
+            [InlineData("json", @"[{
+    name: 'Eric Mill',
+    github: 'konklone',
+},{
+    name: 'Parker Moore',
+    github: 'parkr'
+},{
+    name: 'Liu Fengyun',
+    github: 'liufengyun'
+}]")]
+            [InlineData("csv", @"name,github
+""Eric Mill"",""konklone""
+""Parker Moore"",""parkr""
+""Liu Fengyun"",""liufengyun""")]
+            [InlineData("tsv", @"name	github
+""Eric Mill""	""konklone""
+""Parker Moore""	""parkr""
+""Liu Fengyun""	""liufengyun""")]
+            public void renders_nested_lists(string ext, string fileContent)
+            {
+                fileSystem.AddFile(Path.Combine(dataDirectory, $"members.{ext}"), new MockFileData(fileContent));
+
+                var template = Template.Parse(@"{{ data.members | size }}");
+
+                var hash = Hash.FromAnonymousObject(new
+                {
+                    Data = data
+                });
+
+                var result = template.Render(hash);
+
+                Assert.Equal("3", result.Trim());
+            }
+
+            [Theory]
+            [InlineData("yml", @"dave:
+    name: David Smith
+    twitter: DavidSilvaSmith")]
+            [InlineData("json", @"{
+    dave: {
+        name: 'David Smith',
+        twitter: 'DavidSilvaSmith'
+    }
+}")]
+            //TODO: This is currently not supported. See https://jekyllrb.com/docs/datafiles/#example-accessing-a-specific-author
```

Comment:
Is this planned before merging or not possible without heavy modification?

Reply:
If it can't be part of the released, a note should be added in the [Jekyll differences page](https://github.com/Code52/pretzel/wiki/Jekyll-differences).

Reply:
I'm not even sure if Jekyll behaves this way with csv/tsv files. To be honst I wasn't even sure if CSV/TSV adds any value. I just followed along the samples in Jekyll docs and assumed this seams to work. But yes, let's leave it that way and not it in the differences.

Reply:
But it works fine with yml and json.

Reply:
Oh, ok I was missing the InlineData attributes for yml and json, I understand.
Does it raise an exception with csv / tsv?
If yes it should be explicit.
If not, the note in wiki should be enough, I agree with you about yml / json sufficient for most cases.

Reply:
No it does not raise an exception, it just returns an empty string.

Reply:
The main reason I don't wan't to add this feature for now: I would need to write a own datastructure that act's like an object, a list and a dictionary. Seams overkill to me.
If there is only 1 record in the CSV i return directly the dictionary. If the fieldnames contains dots i convert it into nested dictionaries so it behaves like an object. That works fine for 99% the cases. But in this special case of course there isn't a people key in the dictionary. I thought about adding it into it, but that would mean if there is a people column in the csv there it would conflict.

Reply:
Good to me, a note in the wiki page will be sufficient for now.